blueocean-dashboard/src/main/js/creation/bitbucket/cloud/BbCloudFlowManager.js

```diff
@@ -302,8 +302,21 @@ export default class BbCloudFlowManager extends FlowManager {
     _createPipelineComplete(result) {
         this.outcome = result.outcome;
         if (result.outcome === CreateMbpOutcome.SUCCESS) {
-            this.changeState(STATE.STEP_COMPLETE_SUCCESS);
-            this.pipeline = result.pipeline;
+            if (!this.isStateAdded(STATE.STEP_COMPLETE_MISSING_JENKINSFILE)) {
+                setTimeout(() => this._checkForBranchCreation(result.pipeline.name, true, ({ isFound, hasError, pipeline }) => {
+                    if (!hasError && isFound) {
+                        this._finishListening(STATE.STEP_COMPLETE_SUCCESS);
+                        this.pipeline = pipeline;
+                        this.pipelineName = pipeline.name;
+                    }
+                }), this.redirectTimeout);
+                if (!this.isStateAdded(STATE.STEP_COMPLETE_MISSING_JENKINSFILE)
+                    && !this.isStateAdded(STATE.STEP_COMPLETE_SUCCESS)) {
+                    this.changeState(STATE.PENDING_CREATION_SAVING);
+                    this.pipeline = result.pipeline;
+                    this.pipelineName = result.pipeline.name;
+                }
+            }
```

Comment:
This code block is a bit confusing to me. The request to save completes (presumably w/ status=201). In what situation would the "missing jenkinsfile" state already be added? Also I assumed after a successful save it would enter `PENDING_CREATION_EVENTS` rather than `PENDING_CREATION_SAVING` - as that state is set when the initial save is started.